raiden/network/proxies/token_network.py

```diff
@@ -711,12 +714,62 @@ def update_transfer(
         }
         log.debug('updateNonClosingBalanceProof called', **log_details)
 
+        data_that_was_signed = pack_balance_proof(
+            nonce=nonce,
+            balance_hash=balance_hash,
+            additional_hash=additional_hash,
+            channel_identifier=channel_identifier,
+            token_network_identifier=self.address,
+            chain_id=self.proxy.contract.functions.chain_id().call(),
+        )
+
+        try:
```

Comment:
I would suggest moving this  into its own function and use it here and [here](https://github.com/raiden-network/raiden/pull/2671/files#diff-fe7772da6084a3493b31e88fbf7f6d55R255).

By this I mean from lines 726 until 742. Just to avoid code duplication.

Reply:
This is literally one function call, I can't see how one can make a smaller unit of work.

Reply:
You have the same comment and are catching bare `Exception` in two different places in the code. I suggest the above to keep it in one place.

Reply:
The two handlers behave differently, one returns a ErrorOrResult the other raises an exception. What benefit do you foresee with this change? The only thing I cant think of is changing the exception in the handler at the same time, for that a constant would be better.

Reply:
> What benefit do you foresee with this change? The only thing I cant think of is changing the exception in the handler at the same time, for that a constant would be better.

Sounds fine. It's also fine if you do nothing here. I simply wanted to see if we could avoid code duplication.